tests/test_posixtest.py

```diff
@@ -76,6 +76,12 @@ def get_pthread_tests():
   'test_pthread_rwlock_trywrlock_1_1': 'fails with "main: Unexpected thread state"',
   'test_pthread_spin_destroy_3_1': 'never returns',
   'test_pthread_spin_init_4_1': 'never returns',
+  'test_pthread_attr_setinheritsched_2_3': 'scheduling policy/parameters unsupported',
+  'test_pthread_attr_setinheritsched_2_4': 'scheduling policy/parameters unsupported',
+  'test_pthread_attr_setschedparam_1_3': 'scheduling policy/parameters unsupported',
+  'test_pthread_attr_setschedparam_1_4': 'scheduling policy/parameters unsupported',
+  'test_pthread_attr_setschedpolicy_4_1': 'scheduling policy/parameters unsupported',
+  'test_pthread_getschedparam_1_3': 'scheduling policy/parameters unsupported',
```

Comment:
I guess this the price we pay.  How can we tell if its worth it?

Reply:
Note that this change _might_ suggest that these are the only unsupported pthread tests we are skipping. This is not the case, see for example: https://github.com/emscripten-core/emscripten/pull/13059.

Reply:
Could/should we modify the tests so that they detect the feature and report "unsupported" instead of failing?

Reply:
Good idea, that reminds me of the `xfail` marker of pytest. Is this already supported by the test runner?

Reply:
I was thinking we should special case this.. we would need some way to detect that the posixtest reported unsupported.  Perhaps its as simple as adding:

``
if test in exptected_unsupported:
   self.assertContained('UNSUPPORTED', outout)
else:
   self.assertNotContained('UNSUPPORTED', output)
```

Reply:
Unfortunately, the POSIX test suite sometimes fails with a non-zero exit code when it is actually an unresolved test. See for example:
https://github.com/emscripten-core/posixtestsuite/blob/d8b4410774511596d95f6057abc4c88b1dc63c05/conformance/interfaces/pthread_getschedparam/1-2.c#L42-L47
https://github.com/emscripten-core/posixtestsuite/blob/d8b4410774511596d95f6057abc4c88b1dc63c05/conformance/interfaces/pthread_getschedparam/1-3.c#L106-L109

So asserting that the output contains `UNSUPPORTED` will not always work. Instead, I introduced a `xfail` parameter for the `run_js` and `run_browser` function with commit https://github.com/kleisauke/emscripten/commit/32bdaefec8ab35921fd4e16d129988394fbadf9f, which resolves this in a different way (I'm happy to revisit this if this approach is not desired).

Reply:
Lets propose the xfail thing in seperate PR to keep this one more focused.

(I'm hoping we can switch to `best_exit` and maybe use `expected=NON_ZERO` rather than adding xfail).

Reply:
I reverted the xfail thing for now with commit https://github.com/kleisauke/emscripten/commit/0ac7c324c2f48633dd446211099d1027b90e9ba9.